silkworm/silkrpc/commands/ots_api.cpp

```diff
@@ -249,6 +251,115 @@ boost::asio::awaitable<void> OtsRpcApi::handle_ots_getBlockTransactions(const nl
     co_return;
 }
 
+boost::asio::awaitable<void> OtsRpcApi::handle_ots_getTransactionBySenderAndNonce(const nlohmann::json& request, nlohmann::json& reply) {
+    const auto& params = request["params"];
+    if (params.size() != 2) {
+        const auto error_msg = "invalid ots_getTransactionBySenderAndNonce params: " + params.dump();
+        SILKRPC_ERROR << error_msg << "\n";
+        reply = make_json_error(request["id"], 100, error_msg);
+        co_return;
+    }
+
+    const auto sender = params[0].get<evmc::address>();
+    const auto nonce = params[1].get<uint64_t>();
+
+    auto sender_byte_view = full_view(sender);
+
+    SILKRPC_DEBUG << "sender: " << sender << " nonce: " << nonce << "\n";
+
+    auto tx = co_await database_->begin();
+    ethdb::TransactionDatabase tx_database{*tx};
+    try {
+        auto account_history_cursor = co_await tx->cursor(db::table::kAccountHistoryName);
+        auto account_change_set_cursor = co_await tx->cursor_dup_sort(db::table::kAccountChangeSetName);
+        auto key_value = co_await account_history_cursor->seek(sender_byte_view);
+
+        uint64_t* blocks;
+
+        uint64_t max_block_prev_chunk = 0;
+        roaring::Roaring64Map bitmap;
+
+        while (true) {
+            if (key_value.key.empty() || !key_value.key.starts_with(sender_byte_view)) {
+                auto plane_state_cursor = co_await tx->cursor(db::table::kPlainStateName);
+                auto account_payload = co_await plane_state_cursor->seek(sender_byte_view);
+                auto account = Account::from_encoded_storage(account_payload.value).value();
+
+                if (account.nonce > nonce) {
+                    break;
+                }
+
+                reply = make_json_content(request["id"], nullptr);
+                co_await tx->close();
+                co_return;
+            }
+
+            bitmap = silkworm::db::bitmap::parse(key_value.value);
+            auto max_block = bitmap.maximum();
+            auto block_key{silkworm::db::block_key(max_block)};
+            auto account_payload = co_await account_change_set_cursor->seek_both(block_key, sender_byte_view);
+            account_payload = account_payload.substr(sender_byte_view.length());
+            auto account = Account::from_encoded_storage(account_payload).value();
+
+            if (account.nonce > nonce) {
+                break;
+            }
+
+            max_block_prev_chunk = max_block;
+            key_value = co_await account_history_cursor->next();
+        }
+
+        uint64_t cardinality = bitmap.cardinality();
+        blocks = new uint64_t[cardinality];
+        bitmap.toUint64Array(blocks);
+
+        uint64_t idx = 0;
+        for (uint64_t i = 0; i < cardinality; i++) {
+            auto bn = blocks[i];
```

Comment:
`blocks` should be renamed `account_block_numbers`
`bn` should be renamed `block_number`